rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java

```diff
@@ -2168,19 +2169,31 @@ private Project convertToProject(Map<String, Object> requestBody) {
         ObjectMapper mapper = new ObjectMapper();
         mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
         mapper.registerModule(sw360Module);
+        try {
+            if (requestBody.containsKey(mapOfProjectFieldsToRequestBody.get(Project._Fields.VISBILITY))) {
+                String visibility = (String) requestBody
+                        .get(mapOfProjectFieldsToRequestBody.get(Project._Fields.VISBILITY));
+                requestBody.put(mapOfProjectFieldsToRequestBody.get(Project._Fields.VISBILITY),
+                        visibility.toUpperCase());
+            }
 
-        if (requestBody.containsKey("linkedProjects")) {
-            Map<String, Object> linkedProjects = (Map<String, Object>) requestBody.get("linkedProjects");
-            linkedProjects.entrySet().stream().forEach(entry -> {
-                if (entry.getValue() instanceof String) {
-                    Map<String, Object> projectProjectRelationShip = new HashMap<String, Object>();
-                    projectProjectRelationShip.put("projectRelationship", entry.getValue());
-                    linkedProjects.put(entry.getKey(), projectProjectRelationShip);
-                }
-            });
-
+            if (requestBody.containsKey("linkedProjects")) {
+                Map<String, Object> linkedProjects = (Map<String, Object>) requestBody.get("linkedProjects");
+                linkedProjects.entrySet().stream().forEach(entry -> {
+                    if (entry.getValue() instanceof String) {
+                        Map<String, Object> projectRelationshipMap = new HashMap<>();
+                        projectRelationshipMap.put("projectRelationship", entry.getValue());
+                        linkedProjects.put(entry.getKey(), projectRelationshipMap);
+                    }
+                });
+            }
+            JsonNode jsonNode = mapper.valueToTree(requestBody);
+            return mapper.convertValue(requestBody, Project.class);
+        } catch (IllegalArgumentException e) {
+            System.err.println("Error converting requestBody to Project: " + e.getMessage());
+            System.err.println("Failed requestBody: " + requestBody);
+            throw e;
```

Comment:
Please close the catch section with case changing `if` condition, that is the only modified section of this file. Existing code was not modified in this PR, thus new exception is not expected for entire function.